File: libs/language/src/lib/mapping/__tests__/schema/y.ts
import { z } from 'zod';
import { xSchema } from './x';
import { x1Schema } from './x1';
import { x2Schema } from './x2';
import { xSubscriptionSchema } from './xSubscription';
import { withTypeBlueId } from '../../annotations';

export const ySchema = withTypeBlueId('Y-BlueId')(
  z.object({
    xField: xSchema.optional(),
    x1Field: x1Schema.optional(),
    x2Field: x2Schema.optional(),

    xListField: z.array(xSchema).optional(),
    xMapField: z.map(z.string(), xSchema).optional(),
    x1SetField: z.set(x1Schema).optional(),
    x2MapField: z.map(z.string(), x2Schema).optional(),
    xArrayField: z.array(xSchema).optional(),

    wildcardXListField: z.array(xSchema).optional(),
    subscriptions: z.array(xSubscriptionSchema).optional(),
  })
);

File: references/java-sdk/src/test/java/blue/language/mapping/model/Y.java
package blue.language.mapping.model;

import blue.language.model.TypeBlueId;

import java.util.List;
import java.util.Map;
import java.util.Set;

@TypeBlueId("Y-BlueId")
public class Y {
    public String type;
    public X xField;
    public X1 x1Field;
    public X2 x2Field;
    public List<X> xListField;
    public Map<String, X> xMapField;
    public Set<X1> x1SetField;
    public Map<String, X2> x2MapField;
    public X[] xArrayField;
    public List<? extends X> wildcardXListField;
    public List<XSubscription> subscriptions;

}